src/finch/interface/overrides.py

```diff
@@ -0,0 +1,59 @@
+import operator
+import numpy as np
+from abc import ABC, abstractmethod
+    
+class AbstractOverrideTensor(ABC):
+    @abstractmethod
+    def override_module(self):
+        """Return the module that implements the override logic."""
+        pass
+    
+    def __array_function__(self, func, types, args, kwargs):
+        """Override NumPy functions using the __array_function__ protocol."""
+        # https://numpy.org/neps/nep-0018-array-function-protocol.html
+        func = getattr(self.override_module(), func.__name__)
+        if func is None:
+            return NotImplemented
+        return func(*args, **kwargs)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        """Override NumPy ufuncs using the __array_ufunc__ protocol."""
+        # https://numpy.org/devdocs/user/basics.ufuncs.html#ufuncs-basics
+        # https://numpy.org/devdocs/reference/ufuncs.html#ufuncs-methods
+        ufunc_map = {
+            np.add: operator.add,
+            np.subtract: operator.sub,
+            np.multiply: operator.mul,
+            np.negative: operator.neg,
+            np.positive: operator.pos,
+            np.absolute: operator.abs,
+            np.abs: operator.abs,
+            # Add more ufuncs as needed
+        }
```

Comment:
This will be constructed each time. Let's move it out into the module and make the variable name all caps to represent a constant.